Add isocontour bounding boxes (#2028)
chaos-polymtl/lethe: doc/source/parameters/cfd/post_processing.rst

* ``phase volumes name``: name of the output file containing phase energies from Cahn-Hilliard simulations. The default file name is ``phase_volumes``.
        
Isocontour Bounding Box
~~~~~~~~~~~~~~~~~~~~~~~

> The ``isocontour bounding box`` subsection defines isocontours to be monitored. This is used to get the time evolution of one or multiple isocontours' bounding values:

Reviewer: ```suggestion
The ``isocontour bounding box`` subsection defines isocontours to be monitored. This is used to get the time evolution the isocontours of one or multiple fields:
```

* ``phase volumes name``: name of the output file containing phase energies from Cahn-Hilliard simulations. The default file name is ``phase_volumes``.

Isocontour Bounding Box
~~~~~~~~~~~~~~~~~~~~~~~

> The ``isocontour bounding box`` subsection defines isocontours to be monitored. This is used to get the time evolution of isocontours of one or multiple fields:

Reviewer 2: I feel this sentence is misleading, do we output/track the actual interface (full interface information throught time) or simply the bounding box coordinates through time? I'm guessing its the coordinates that you can then use this efficiently with camera settings in paraview?
```suggestion
The ``isocontour bounding box`` subsection defines isocontours to be monitored. This is used to get the time evolution of the bounding box of isocontours of one or multiple fields to better help with its visualisation:
```

Author: I hadn't even though about paraview. It's really for post-processing and avoid going through python later. It can help get heights, depths, or lengths when you already know what values interest you before running the simulation. 
I'll leave off the "to better help with its visualisation" and add a sentence explaining this better. :)

  - In 2D: :math:`x_\mathrm{min}`, :math:`x_\mathrm{max}`, :math:`y_\mathrm{min}`, and :math:`y_\mathrm{max}`;
  - In 3D: :math:`x_\mathrm{min}`, :math:`x_\mathrm{max}`, :math:`y_\mathrm{min}`, :math:`y_\mathrm{max}`, :math:`z_\mathrm{min}`, and :math:`z_\mathrm{max}`.

* ``number of isocontour bounding boxes``: number of monitored isocontours.

> * ``variable``: variable(s) of monitored isocontour(s). For multiple isocontours, the different variables must be separated with a comma (e.g., ``set variables = phase, temperature, temperature``).

Reviewer 2: ```suggestion
* ``variable``: variable(s) of monitored isocontour(s). For multiple isocontours, the different variables must be separated with a comma (e.g., ``set variable = phase, temperature, temperature``).
```

  .. note::
    At the moment, only isocontours on the ``temperature`` and ``phase`` (:doc:`CLS<../../../theory/multiphase/cfd/cls>` phase indicator) are implemented.

* ``isovalue``: isovalue(s) of monitored isocontour(s). For multiple isocontours, the different isovalues must be separated with a comma (e.g., ``set isovalue = 0.5, 300, 500``).

> * ``bounding box filename``: filename(s) for outputted isocontour(s). For multiple isocontours, the different filenames must be separated with a comma (e.g., ``set isovalue = interface_bounding_box, solidus_bounding_box, liquidus_bounding_box``)

Reviewer 2: ```suggestion
* ``bounding box filename``: filename(s) for outputted isocontour(s). For multiple isocontours, the different filenames must be separated with a comma (e.g., ``set bounding box filename = interface_bounding_box, solidus_bounding_box, liquidus_bounding_box``)
```

Isocontour Bounding Box
~~~~~~~~~~~~~~~~~~~~~~~
The ``isocontour bounding box`` subsection defines isocontours to be monitored. This is used to get the time evolution of the bounding boxes of isocontours of one or multiple fields:
  - In 2D: :math:`x_\mathrm{min}`, :math:`x_\mathrm{max}`, :math:`y_\mathrm{min}`, and :math:`y_\mathrm{max}`;
  - In 3D: :math:`x_\mathrm{min}`, :math:`x_\mathrm{max}`, :math:`y_\mathrm{min}`, :math:`y_\mathrm{max}`, :math:`z_\mathrm{min}`, and :math:`z_\mathrm{max}`.

This feature can be used to evaluate heights, depths, or lengths when quantities of interests are known prior to running simulations. Each isocontour's bounding values in function of time is stored in a separate ``.dat`` file with named as specified with ``bounding box filename``.

.. warning::
  If, during simulation, a tracked isocontour is not detected within the simulated domain at a given simulation time, no entry will be made for that time. This means that if an empty ``.dat`` file is outputted, the tracked isocontour was not present in the domain during the simulated time.

* ``number of isocontour bounding boxes``: number of monitored isocontours.

> * ``variable``: variable(s) of monitored isocontour(s). For multiple isocontours, the different variables must be separated with a comma (e.g., ``set variable = phase, temperature, temperature``).

Reviewer 3: ```suggestion
* ``variable``: variable(s) used to define the monitored isocontour(s). Each entry corresponds to one isocontour. When multiple isocontours are defined, the variables must follow the same order as the isovalues and be separated by commas (e.g., ``set variables = phase, temperature, temperature``).
```
